File: sources/Shared.Command/Models/MessageModel.cs
namespace Lewio.CashFlow.Shared;
#nullable disable

public partial struct MessageModel
{
   public string Text { get; set; }
   public TypeEnum Type { get; set; }
}

partial struct MessageModel
{
   public enum TypeEnum : short { Message = 0, Warning = 1, Error = 2 }
}

partial struct MessageModel
{

   public static MessageModel CreateMessage(string text) =>
      Create(text, TypeEnum.Message);

   public static MessageModel CreateWarning(string text) =>
      Create(text, TypeEnum.Warning);

   public static MessageModel CreateError(string text) =>
      Create(text, TypeEnum.Error);

   internal static MessageModel Create(string text, TypeEnum type) =>
      new MessageModel
      {
         Text = text,
         Type = type
      };

}

File: sources/Frontend/src/app/components/messages/models/message.model.ts
import { MessageTypeEnum } from "..";

export class MessageModel {
   Text: string = '';
   Type: MessageTypeEnum = MessageTypeEnum.Message;

   public static Parse(data: MessageModel): MessageModel {
      return Object.assign(new MessageModel, data);
   }

   public static CreateWarning(text: string) {
      return MessageModel.Create(text, MessageTypeEnum.Warning);
   }

   public static CreateError(text: any) {
      return MessageModel.Create(text, MessageTypeEnum.Error);
   }

   private static Create(text: string, type: MessageTypeEnum) {
      const result: MessageModel = {
         Text: text,
         Type: type
      };
      return result;
   }

}
